addons/mail/models/mail_message.py

```diff
@@ -90,9 +90,21 @@ def default_get(self, fields):
     preview = fields.Char(
         'Preview', compute='_compute_preview',
         help='The text-only beginning of the body used as email preview.')
+    link_preview_message_ids = fields.One2many(
+        "mail.link.preview.message",
+        "message_id", domain=[("is_hidden", "=", False)], groups="base.group_erp_manager")
     link_preview_ids = fields.One2many(
```

Comment:
This is actually a m2m using ``mail_link_preview_message`` as intermediate table, isn't it ? This should be defined accordingly. We have this pattern in other modules (e.g. slide channel members, ...).

Reply:
This is not working since we must filter the link_preview using the is_hidden field.
The ORM doesn't produce a valid query for this, as it passes the condition to a subquery that doesn't filter them properly.

```sql
SELECT
    mail_link_preview_message.message_id,
    mail_link_preview_message.link_preview_id
FROM
    mail_link_preview
    JOIN mail_link_preview_message ON
            mail_link_preview_message.link_preview_id
            = mail_link_preview.id
WHERE
    mail_link_preview.id
    IN (
            SELECT
                mail_link_preview_message.link_preview_id
            FROM
                mail_link_preview_message
            WHERE
                mail_link_preview_message.is_hidden
                IS NOT true
        )
    AND mail_link_preview_message.message_id IN (26, 25, 24, 23, 9, 8, 7, 6, 1)
ORDER BY
    mail_link_preview.id;
```

Reply:
Ah yes, if you have a field to filter in the middle table, the ORM does not take it into account, it is not working like that. Here you have to defined a computed Many2many (not One2many). It does not change much, but it behaves like a m2m (as you don't have a unique inverse relationship).

Reply:
Still to define as a compute many2many, not one2many.